ledger/participant-integration-api/src/main/scala/platform/configuration/IndexServiceConfig.scala

```diff
@@ -46,3 +58,41 @@ object IndexServiceConfig {
   val DefaultInMemoryFanOutThreadPoolSize: Int = 16
   val PreparePackageMetadataTimeOutWarning: FiniteDuration = FiniteDuration(1, "second")
 }
+
+// TODO etq: Take care what config key names will get exposed in the user-facing participant config
+case class TransactionsFlatStreamReaderConfig(
+    maxIdsPerIdPage: Int = 20000,
+    maxPagesPerIdPagesBuffer: Int = 1,
+    maxWorkingMemoryInBytesForIdPages: Int = 100 * 1024 * 1024,
+    maxPayloadsPerPayloadsPage: Int = 1000, // eventsPageSize
+    maxParallelIdCreateQueries: Int = 4,
+    maxParallelIdConsumingQueries: Int = 4,
+    // TODO etq: This must be a power of 2
+    maxParallelPayloadCreateQueries: Int = 2,
+    // TODO etq: This must be a power of 2
+    maxParallelPayloadConsumingQueries: Int = 2,
+    maxParallelPayloadQueries: Int = 2,
+    payloadProcessingParallelism: Int = 8,
+)
+object TransactionsFlatStreamReaderConfig {
+  def default: TransactionsFlatStreamReaderConfig = TransactionsFlatStreamReaderConfig()
+}
+
+// TODO etq: Take care what config key names will get exposed in the user-facing participant config
+case class TransactionsTreeStreamReaderConfig(
```

Comment:
I wonder why we couldn't have just one case class for both stream types?

Reply:
Tree tx streaming config has some exclusive settings because it's using more (all) id filter tables for fetching ids and it also fetches non-consuming events.
The settings that are named the same between those two configs should have analogous meaning, but I wouldn't go for merging them as one might one be able to independently tune tx and flat streams.

If we were to think about simplifying & merging those configs, I think that should be handled Canton where they'll be mapping to settings that are exposed to the end users